src/redux/book/booksSlice.js

```diff
@@ -0,0 +1,20 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+    books: [],
+};
+
+const bookSlice = createSlice({
+    name: 'book',
+    initialState,
+    reducers: {
+        addBook: (state, action) => {
+            state.books.push(action.payload)
+        },
+        removeBook: (state, payload) => {
+            state.books.filter((item) => item.id === payload.id);
+        },
```

Comment:
There is a mistake in this piece of code. You are directly passing in the `payload` instead of `action`. Then you can access `payload` on `action`, but you can't do it directly. Kindly fix this issue. In the `addBook` reducer, you have done it the right way.